src/types/Currency.sol

```diff
@@ -39,14 +39,14 @@ library CurrencyLibrary {
     error Wrap__ERC20TransferFailed(address token, bytes reason);
 
     /// @notice A constant to represent the native currency
-    Currency public constant NATIVE = Currency.wrap(address(0));
```

Comment:
I also like Native fwiw but we wanted to consolidate the isNative() and isZero() paths and there was a an issue open zeppelin found that makes a good case for when isNative() can be confusing on chains where the native token is also an erc20 token (ie Celo).

Reply:
If you have an alternative suggestion feel free to add it. We just don't want to have both isNative() and isZero() functions

Reply:
My suggestion is to remove `isZero` instead of `isNative`. I understand there are chain-specific native token precompiles though. It will be slightly unclear either way, but not to me, so I am not a great judge of this.